core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/GPUTests.scala

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.nbtest
+
+import com.microsoft.azure.synapse.ml.build.BuildInfo
+import com.microsoft.azure.synapse.ml.core.env.FileUtilities
+import com.microsoft.azure.synapse.ml.nbtest.DatabricksUtilities._
+
+import java.io.File
+import java.io.PrintWriter
+import scala.collection.mutable.ListBuffer
+import scala.io.Source
+
+class GPUTests extends DatabricksTestHelper {
+
+  val horovodInstallationScript: File = FileUtilities.join(
+    BuildInfo.baseDirectory.getParent, "deep-learning",
+    "src", "main", "python", "horovod_installation.sh").getCanonicalFile
+  uploadFileToDBFS(horovodInstallationScript, "/FileStore/horovod/horovod_installation.sh")
+  val clusterId: String = createClusterInPool(GPUClusterName, AdbGpuRuntime, GpuPoolId, GPUInitScripts)
+  GPUNotebooks.foreach(
+    file => {
+      val tmpFile = new File("tmp.txt")
+      val w = new PrintWriter(tmpFile)
+      Source.fromFile(file).getLines().map {
+        _.replaceAll("CUR_VERSION", BuildInfo.version)
+      }.foreach(x => w.println(x))
+      w.close()
+      tmpFile.renameTo(file)
+    }
+  )
```

Comment:
What is this code doing exactly?

Reply:
In the notebook, the first cell is this `%pip install https://mmlspark.blob.core.windows.net/pip/CUR_VERSION/synapseml_deep_learning-0.10.0.dev1-py2.py3-none-any.whl --force-reinstall --no-deps` so I add this extra piece to replace CUR_VERSION to the correct build version, then it will install the wheel package on all nodes in databricks. FYI I have tried excluding this pip install and just install our jar package, unfortunately that doesn't work with horovod as it seems can't find the correct module where dl code lives. I'm not sure why though, but installing the wheel package should be enough for deep vision usage.

Reply:
you might be able to use synapse.ml.core.__scala_version__ here instead of code generation/modification that is hard for the user to do